### ISSUE TITLE
`project_name` example is incorrect, and probably not how macros should be dispatched now

### ISSUE THREAD

**Issue by rlh1994** [fishtown-analytics/docs.getdbt.com#3529]
### Contributions

- [X] I have read the contribution docs, and understand what's expected of me.

### What page(s) or areas on docs.getdbt.com are affected?

https://docs.getdbt.com/reference/dbt-jinja-functions/project_name

### What changes are you suggesting?

The `project_name` value is now accessed directly, rather than as part of the `context` variable, the example is no longer valid.

In general this example also doesn't align with how dbt suggests to dispatch and use project vs package macros anymore either, a new example may be more relevant.

### Additional information

Using dbt version 1.5.1

**Comment by runleonarun** [comment]
@joellabes can you verify this and provide a new example if necessary?

**Comment by joellabes** [comment]
@rlh1994 is right! You can access project_name directly: `{{ project_name }}`. 

I followed the [file's git history](https://github.com/dbt-labs/docs.getdbt.com/commits/c5ae1f91c31542dd90241a815a4a250672924cb5/website/docs/reference/dbt-jinja-functions/project_name.md) all the way [back to its original creation](https://github.com/dbt-labs/docs.getdbt.com/commits/5c9efcc1b3c5ac8b1579db52d2d5bb837e0ed655/website/docs/docs/writing-code-in-dbt/jinja-context/project_name.md?browsing_rename_history=true&new_path=website/docs/reference/dbt-jinja-functions/project_name.md&original_branch=c5ae1f91c31542dd90241a815a4a250672924cb5) and it hasn't had any substantive changes since it was created in March 2020, which predates `adapter.dispatch`. 

So I think we can delete this altogether. The only thing I couldn't easily find was any documentation that `{{ project_name }}` itself was available as a variable - that might be a gap that needs filled, but I didn't look beyond using the Algolia search

**Comment by rlh1994** [comment]
@joellabes I couldn't find anything either, I ended up finding the right method from [this PR](https://github.com/dbt-labs/dbt-core/pull/2085), which also helpfully tells you when/where each context is available from and what form, maybe a little overkill for basic users but would be nice to have that documented somewhere?

**Comment by joellabes** [comment]
Makes sense! Are you just thinking these three from [here](https://github.com/dbt-labs/dbt-core/pull/2085#:~:text=High%2Dlevel%20context%20behavior%20changes%3A)? 

>'dbt_version' is always available
>'target' is available as long as the profile has been parsed
>'project_name' is available in: query headers, hooks, models, and macro execution

**Comment by rlh1994** [comment]
Yep those are the ones!

**Comment by joellabes** [comment]
@runleonarun sounds like the new ticket is replace https://docs.getdbt.com/reference/dbt-jinja-functions/project_name with an explanation of those three variables listed above and when they're accessible

**Comment by runleonarun** [comment]
Thanks so much for all the legwork on this @joellabes and @rlh1994! I added priority for the team and we will address this as soon as we can! 

We are happy to review a PR if someone opened a PR capturing this change.
